lighthouse-core/audits/redirects.js

```diff
@@ -20,68 +20,92 @@ class Redirects extends Audit {
   }
 
   /**
-   * @param {!Artifacts} artifacts
-   * @return {!AuditResult}
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
    */
-  static audit(artifacts) {
-    const data = {URL: artifacts.URL, devtoolsLog: artifacts.devtoolsLogs[Audit.DEFAULT_PASS]};
-    return artifacts.requestMainResource(data)
-      .then(mainResource => {
-        // redirects is only available when redirects happens
-        const redirectRequests = Array.from(mainResource.redirects || []);
-
-        // add main resource to redirectRequests so we can use it to calculate wastedMs
-        redirectRequests.push(mainResource);
-
-        let totalWastedMs = 0;
-        const pageRedirects = [];
-
-        // Kickoff the results table (with the initial request) if there are > 1 redirects
-        if (redirectRequests.length > 1) {
-          pageRedirects.push({
-            url: `(Initial: ${redirectRequests[0].url})`,
-            wastedMs: 0,
-          });
-        }
-
-        for (let i = 1; i < redirectRequests.length; i++) {
-          const initialRequest = redirectRequests[i - 1];
-          const redirectedRequest = redirectRequests[i];
-
-          const wastedMs = (redirectedRequest.startTime - initialRequest.startTime) * 1000;
-          totalWastedMs += wastedMs;
-
-          pageRedirects.push({
-            url: redirectedRequest.url,
-            wastedMs,
-          });
-        }
-
-        const headings = [
-          {key: 'url', itemType: 'text', text: 'Redirected URL'},
-          {key: 'wastedMs', itemType: 'ms', text: 'Time for Redirect', granularity: 1},
-        ];
-        const summary = {wastedMs: totalWastedMs};
-        const details = Audit.makeTableDetails(headings, pageRedirects, summary);
-
-        return {
-          // We award a passing grade if you only have 1 redirect
-          score: redirectRequests.length <= 2 ? 1 : UnusedBytes.scoreForWastedMs(totalWastedMs),
-          rawValue: totalWastedMs,
-          displayValue: Util.formatMilliseconds(totalWastedMs, 1),
-          extendedInfo: {
-            value: {
-              wastedMs: totalWastedMs,
-            },
-          },
-          details,
-        };
+  static async audit(artifacts, context) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const simulatorData = {devtoolsLog, settings: context.settings};
+    const simulator = await artifacts.requestLoadSimulator(simulatorData);
+
+    // TODO(phulce): split MetricComputationInput type and avoid this cast
+    const metricSettings = /** @type {LH.Config.Settings} */ ({throttlingMethod: 'simulate'});
+    const metricComputationData = {trace, devtoolsLog, simulator, settings: metricSettings};
+
+    const mainResource = await artifacts.requestMainResource({URL: artifacts.URL, devtoolsLog});
+    const metricResult = /** @type {LH.Artifacts.LanternMetric} */ (
```

Comment:
any reason to not just ask for `artifacts.requestLanternInteractive()` directly?

Reply:
yeah I suppose at this point it's not that different 👍 done